test/integration/api/forms/draft.js

```diff
@@ -1401,15 +1401,75 @@ describe('api: /projects/:id/forms (drafts)', () => {
               should.not.exist(body.details);
             }))));
 
-      it('should reject if the user cannot modify', testService((service) =>
-        service.login('alice', (asAlice) =>
-          asAlice.post('/v1/projects/1/forms/simple/draft')
+      const withRole = role => [
+        `has role "${role}"`,
+        (service, asAlice) => service.login('chelsea', async (asChelsea) => {
+          await asChelsea.get('/v1/users/current')
+            .expect(200)
+            .then(({ body }) => body)
+            .then((chelsea) =>
+              asAlice.post(`/v1/projects/1/assignments/${role}/${chelsea.id}`)
+                .expect(200));
+          return asChelsea;
+        }),
+      ];
+
+      const noExtraRoles = () => [
+        'does not have project-specific permissions',
+        (service) => service.login('chelsea'),
+      ];
+
+      [
+        noExtraRoles(),
+        withRole('app-user'),
+        withRole('formfill'),
+        withRole('formview'),
+        withRole('pub-link'),
+        withRole('viewer'),
+      ].forEach(([ description, setupChelsea ]) => {
+        it(`should reject if the user ${description}`, testService(async (service) => {
+          const asAlice = await service.login('alice');
+          await asAlice.post('/v1/projects/1/forms/simple/draft')
             .send(testData.forms.simple)
             .set('Content-Type', 'application/xml')
-            .expect(200)
-            .then(() => service.login('chelsea', (asChelsea) =>
-              asChelsea.get('/v1/projects/1/forms/simple/draft')
-                .expect(403))))));
+            .expect(200);
+
+          const asChelsea = await setupChelsea(service, asAlice);
+          return asChelsea.get('/v1/projects/1/forms/simple/draft')
+            .expect(403);
```

Comment:
Could you also test that Chelsea can't access the form itself? In the case where a form only has a draft and no published version, /v1/projects/1/forms/simple and /v1/projects/1/forms/simple/draft return pretty much the same thing. In contrast, in the next test, Chelsea should be able to access the form, just not the draft.

```js
await asChelsea.get('/v1/projects/1/forms/simple')
  .expect(403);
```

Reply:
@matthew-white is this still required?